### ISSUE TITLE
save and load TensorFlowV2Classifier from .ckpt file

### ISSUE THREAD

**Issue by le000043** [Trusted-AI/adversarial-robustness-toolbox#1019]
How can we save and load TensorFlowV2Classifier from .ckpt file?

**Comment by beat-buesser** [comment]
Hi @le000043 Thank you very much for using ART! If I understand your example correctly, you would have to load the model from the checkpoint using TensorFlow and the use the loaded model to create a new instance of `TensorFlowV2Classifier`.

**Comment by le000043** [comment]
Hello,
thank you for getting back to me. How exactly do we save the TensorFlowV2Classifier in the first place?

**Comment by beat-buesser** [comment]
Currently you have to save the model using TensorFlow functions.

**Comment by le000043** [comment]
Can you point me to where how exact we are using TensorFlow functions to save TensorFlowV2Classifier ?

**Comment by beat-buesser** [comment]
Could you please share an example code of your applicaition?

**Comment by le000043** [comment]
`tf2_model = TensorFlowModel()
  tf2_classifier = TensorFlowV2Classifier(
      model=tf2_model,
      loss_object=loss_object,
      train_step=train_step,
      nb_classes=10,
      input_shape=input_shape,
      clip_values=(0, 1),
    )
    # create data generator
    #######################################################
    from keras.preprocessing.image import ImageDataGenerator
    datagen = ImageDataGenerator(width_shift_range=0.1, height_shift_range=0.1, horizontal_flip=True)
    datagen.fit(x_train_01)
    # prepare iterator
    it_train = datagen.flow(x_train_01, y_train_01, batch_size=128,seed = 1) # old: 64
    from art.data_generators import KerasDataGenerator
    data_generator = KerasDataGenerator(
                    iterator=it_train, size=x_train_01.shape[0], batch_size=128, # old: 64
                )
    #######################################################
    steps = int(x_train_01.shape[0] / 128)
    print("fitting tf2_classifier 00")
    tf2_classifier.fit_generator(data_generator, steps_per_epoch=steps, nb_epochs=50, verbose=0) # validation_data=(x_test, y_test)`
    it would be greatly appreciated if you can help me out

**Comment by beat-buesser** [comment]
Ok, I see. I think the best option for you is to use methods provided by TensorFlow to save the model `TensorFlowV2Classifier` by accessing it at `tf2_classifier.model`, because `TensorFlowV2Classifier` itself does not have a `save` method implemented. You can later reload the model and recreate a new instance of `TensorFlowV2Classifier`.

You can also train the TensorFlow model outside of `TensorFlowV2Classifier` and provide the trained model to `TensorFlowV2Classifier`, this is actually the preferred approach as it allows you to use all training features of TensorFlow.
